api/secrets/secret_ref.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
```

Comment:
Making me think about our package naming strategy, plural vs singular: ```secrets``` feels more natural here than ```secret```. Do we leave this to the individual SDK implementation?

Reply:
I think the only one we haven't done is storage for pluralization in most of the sdks. It doesn't particularly matter for go, but package names should match their parent directory for easier discoverability.

Reply:
In the Java SDK I am not using pluralization for package names but that is a Java convention.